mm2src/coins/eth/web3_transport.rs

```diff
@@ -130,24 +141,75 @@ impl Transport for Web3Transport {
     #[cfg(not(target_arch = "wasm32"))]
     fn send(&self, _id: RequestId, request: Call) -> Self::Out {
         Box::new(
-            send_request(request, self.uris.clone(), self.event_handlers.clone())
-                .boxed()
-                .compat(),
+            send_request(
+                request,
+                self.nodes.clone(),
+                self.event_handlers.clone(),
+                self.gui_auth_validation_generator.clone(),
+            )
+            .boxed()
+            .compat(),
         )
     }
 
     #[cfg(target_arch = "wasm32")]
     fn send(&self, _id: RequestId, request: Call) -> Self::Out {
-        let fut = send_request(request, self.uris.clone(), self.event_handlers.clone());
+        let fut = send_request(
+            request,
+            self.nodes.clone(),
+            self.event_handlers.clone(),
+            self.gui_auth_validation_generator.clone(),
+        );
         Box::new(SendFuture(Box::pin(fut).compat()))
     }
 }
 
+/// Generates a signed message and inserts it into request
+/// payload if gui_auth is activated. Returns false on errors.
+fn handle_gui_auth_payload_if_activated(
+    gui_auth_validation_generator: &Option<GuiAuthValidationGenerator>,
+    node: &Web3TransportNode,
+    request: &Call,
+    errors: &mut Vec<String>,
+) -> Result<String, bool> {
+    if !node.gui_auth {
+        return Err(true);
+    }
+
+    let generator = match gui_auth_validation_generator.clone() {
+        Some(gen) => gen,
+        None => {
+            errors.push(ERRL!("GuiAuthValidationGenerator is not provided for {:?} node", node));
```

Comment:
We're trying to replace `try_s`, `ERRL` macros here https://github.com/KomodoPlatform/atomicDEX-API/issues/1250
Also I'd suggest refactoring this function that way:
* `handle_gui_auth_payload_if_activated` returns `MmResult<Option<String>, web3::Error>` (`web3::Error` or `Web3RpcError` or another)
* `handle_gui_auth_payload_if_activated` doesn't take `errors` argument
* The function that calls `handle_gui_auth_payload_if_activated` should check if it fails and pushes the error to the `errors` container by itself.

Reply:
Used `Web3RpcError` as an error type instead of `String`